shpc/main/modules/module.py

```diff
@@ -97,6 +86,49 @@ def container_path(self):
             )
         return self._container_path
 
+    def add_local_container(self, container_image):
+        """
+        Set the module container to be a local container (copied over)
+        """
+        basename = os.path.basename(container_image)
+        container_dest = os.path.join(self.container_dir, basename)
+        if not os.path.exists(self.container_dir):
+            utils.mkdir_p(self.container_dir)
+        if not os.path.exists(container_dest):
+            shutil.copyfile(container_image, container_dest)
+        self._container_path = container_dest
+
+    def _add_config_container(self):
+        """
+        Set the module container to be the one defined by the config.path
+        """
+        self._container_path = os.path.join(self.config.entry.dirname, self.config.path)
+
+        if not os.path.exists(self._container_path):
+            logger.exit(
+                "Expected container defined by path %s not found in %s."
+                % (self.config.path, self.config.entry.dirname)
+            )
+        container_dest = os.path.join(self.container_dir, self.config.path)
+
+        # Note that here we are *duplicating* the container, assuming we
+        # cannot use a link, and the registry won't be deleted but the
+        # module container might!
+        if not os.path.exists(container_dest):
+            shutil.copyfile(self._container_path, container_dest)
```

Comment:
I'm now noticing that the copy strategy was already in use. This needs to be the same implementation as in `add_local_container`. Could we perhaps change the logic in `add_container` to:
1. If `container_image` is not set, but `self.config.path` is: set `container_image` to `os.path.join(self.config.entry.dirname, self.config.path)` (line 105)
2. Check the file referred to by `container_image` exists
3. Do the copy/symlinking

Reply:
Why not just use the original path that is provided? it's assuming the user has the container already where he/she wants it.

Reply:
Even simpler :)